src/contexts/application-context/index.tsx

```diff
@@ -0,0 +1,64 @@
+import { createContext, useEffect, useState } from 'react';
+import type { ApplicationContextParams, ApplicationProviderProps } from './types';
+import useCookie from 'react-use-cookie';
+
+export const ApplicationContext = createContext<ApplicationContextParams>({
```

Comment:
What do you think about renaming this context to something like `SessionContext`? Just to make it more specific to the temporary token logic and avoid having this context getting bigger with more stuff, `ApplicationContext` sounds too generic

Reply:
ok so `SessionContext`